### PR TITLE
importer class for Australian Cultural Heritage sites

### PR DESCRIPTION
**Pull request by lokal-profil**
TODO: Proper sources

* Preview at: [/COH/Australia/preview](https://www.wikidata.org/wiki/User:André_Costa_(WMSE)/COH/Australia/preview)
* Mapping at:  [/COH/Australia](https://www.wikidata.org/wiki/User:André_Costa_(WMSE)/COH/Australia)

Task: [T174320](https://phabricator.wikimedia.org/T174320)

### CODE REVIEW

**Comment by Vesihiisi** [comment]
On descriptions: Is there a consensus about adding the country name, if an administrative unit is also present? I'd probably write "$item in $state, Australia", but I'm not 100% sure in this particular case since the strings could end up too long.

File: COH_Australia/Australia_importer.py [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+            'indigenous': self.wd.QtoItemPage('Q38048771'),
+            'historic': self.wd.QtoItemPage('Q38048707'),
+            'natural': self.wd.QtoItemPage('Q38048753')
+        }
+
+    def make_states_map(self):
+        """
+        Retrieve the state/territory mappings from Wikidata.
+
+        Also tries to makes up stuff for the EXT and OS codes.
```
Comment (Vesihiisi) [inline COH_Australia/Australia_importer.py]: "tries to makes up stuff" -> "tries to match items"?

File: COH_Australia/Australia_importer.py [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+
+        # P969: address
+        if ',' in data['address']:
+            protoclaims['P969'] = WdS.Statement(data['address'])
+
+        # P276: place
+        # format for address is "street, town STATE_ISO"
+        # might be possible to match town to a wikidata entry?
+        # but how to ensure the correct state?
+        pywikibot.warning('You never implemented P276')
```
Comment (Vesihiisi) [inline COH_Australia/Australia_importer.py]: But there are P276 in the preview?

File: COH_Australia/Australia_importer.py [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+            else:
+                result[key] = row
+
+        if local_log:
+            log[filename] = local_log
+        return result
+
+
+def handle_dupe(new_entry, old_entry, log, place_id):
+    """Handle duplicate entries in csv by removing conflicting fields to."""
```
Comment (Vesihiisi) [inline COH_Australia/Australia_importer.py]: "to"...?

**Comment by lokal-profil** [comment]
On description: I figured that since every item has P17 skipping country name in the description should be ok. I've left place and state since these are not guaranteed to have been parsed into statements

File: COH_Australia/Australia_importer.py (hunk 1 above)
Comment (lokal-profil) [reply COH_Australia/Australia_importer.py]: What does "make up stuff"  not sounds serious ;)
Done

File: COH_Australia/Australia_importer.py (hunk 2 above)
Comment (lokal-profil) [reply COH_Australia/Australia_importer.py]: Looks like my last commit was never pushed. Will make sure it comes with the review follow-up

**Comment by Vesihiisi** [comment]
I get this, as I said, I'm not 100% sure -- I think I'd add the country if the desc doesn't end up too long. So this is okay... and also I'm hoping for a future where descriptions are abolished, but that's another thing.

File: COH_Australia/Australia_importer.py (hunk 3 above)
Comment (lokal-profil) [reply COH_Australia/Australia_importer.py]: done

File: COH_Australia/Australia_importer.py (hunk 1 above)
Comment (Vesihiisi) [reply COH_Australia/Australia_importer.py]: I think it implies inventing something out of thin air... But what we're doing here is straightforward matching.

**Review by Vesihiisi — APPROVED** [review]

**Pull request merged**